File: snippets/insert/rest-exec.go
package main

import (
	"fmt"
	"io/ioutil"
	"log"
	"net/http"
	"net/url"
)

func main() {
	u, err := url.Parse("http://localhost:9000")
	checkErr(err)

	u.Path += "exec"
	params := url.Values{}
	params.Add("query", `
	  INSERT INTO
	    trades
	  VALUES(
	    to_timestamp('2020-10-09T00:00:00', 'yyyy-MM-ddTHH:mm:ss'),
	    123456
	  );
        `)
	u.RawQuery = params.Encode()
	url := fmt.Sprintf("%v", u)

	res, err := http.Get(url)
	checkErr(err)

	defer res.Body.Close()

	body, err := ioutil.ReadAll(res.Body)
	checkErr(err)

	log.Println(string(body))
}

func checkErr(err error) {
	if err != nil {
		panic(err)
	}
}

File: snippets/insert/rest-exec.js
const fetch = require("node-fetch")
const qs = require("querystring")

const HOST = "http://localhost:9000"

async function run() {
  try {
    const queryData = {
      query: `
        INSERT INTO
          trades
        VALUES(
          to_timestamp('2020-10-09T00:00:00', 'yyyy-MM-ddTHH:mm:ss'),
          123456
        );
      `,
    }

    const response = await fetch(`${HOST}/exec?${qs.encode(queryData)}`)
    const json = await response.json()

    console.log(json)
  } catch (error) {
    console.log(error)
  }
}

run()
